Code does not pass tests on Windows

Code does not pass the [ConfigTest.java](https://github.com/RAR1741/RA22_RobotCode/blob/master/src/test/java/config/ConfigTest.java) tests on Windows, making it impossible to build the robot code on Windows.

<details>
<summary>Executing task: gradlew build</summary>
<br>

```java
> Executing task: gradlew build   -Dorg.gradle.java.home="C:\Users\Public\wpilib\2022\jdk" <

> Task :test

config.ConfigTest > noConfigTest PASSED

config.ConfigTest > configLoadTest FAILED
    java.lang.AssertionError at ConfigTest.java:19

config.ConfigTest > configDefaultValueTest PASSED

config.ConfigTest > configTypeTest FAILED
    org.junit.ComparisonFailure at ConfigTest.java:29

config.ConfigTest > configurableTest FAILED
    java.lang.AssertionError at ConfigTest.java:64

config.ConfigTest > configCommentTest FAILED
    java.lang.AssertionError at ConfigTest.java:43

6 tests completed, 4 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///D:/Robotics/RedAlert/RA22_RobotCode/build/reports/tests/test/index.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
7 actionable tasks: 2 executed, 5 up-to-date
The terminal process "cmd.exe /d /c gradlew build   -Dorg.gradle.java.home="C:\Users\Public\wpilib\2022\jdk"" terminated with exit code: 1.
```

</details>

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -9,9 +9,9 @@ on:
 jobs:
   the_code_actually_works:
     runs-on: ${{ matrix.os }}
-      strategy:
-        matrix:
-          os: [ubuntu-latest, windows-latest, macos-latest]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
       - name: Pull the latest code

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -8,7 +8,11 @@ on:
 
 jobs:
   the_code_actually_works:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest, macos-latest]
+
     steps:
       - name: Pull the latest code
         uses: actions/checkout@v2

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -9,9 +9,9 @@ on:
 jobs:
   the_code_actually_works:
     runs-on: ${{ matrix.os }}
-        strategy:
-            matrix:
-                os: [ubuntu-latest, windows-latest, macos-latest]
+      strategy:
+        matrix:
+          os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
       - name: Pull the latest code

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -36,7 +36,6 @@ jobs:
 
       - name: Upload CodeCov Report
         run: curl -s https://codecov.io/bash | bash
-        # run: bash <(curl -s https://codecov.io/bash)
 
       - name: Upload CodeClimate Report
         uses: paambaati/codeclimate-action@v3.0.0

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -35,7 +35,8 @@ jobs:
         run: ./gradlew clean build jacocoTestReport
 
       - name: Upload CodeCov Report
-        run: bash <(curl -s https://codecov.io/bash)
+        run: curl -s https://codecov.io/bash | bash
+        # run: bash <(curl -s https://codecov.io/bash)
 
       - name: Upload CodeClimate Report
         uses: paambaati/codeclimate-action@v3.0.0

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -9,9 +9,9 @@ on:
 jobs:
   the_code_actually_builds:
     runs-on: ${{ matrix.os }}
-        strategy:
-            matrix:
-                os: [ubuntu-latest, windows-latest, macos-latest]
+      strategy:
+        matrix:
+          os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -9,9 +9,9 @@ on:
 jobs:
   the_code_actually_builds:
     runs-on: ${{ matrix.os }}
-      strategy:
-        matrix:
-          os: [ubuntu-latest, windows-latest, macos-latest]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/gradle.yml
+++ b/.github/workflows/gradle.yml
@@ -8,7 +8,10 @@ on:
 
 jobs:
   the_code_actually_builds:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
       - uses: actions/checkout@v2

--- a/src/test/java/config/ConfigTest.java
+++ b/src/test/java/config/ConfigTest.java
@@ -1,12 +1,13 @@
 package config;
 
-import frc.robot.config.Config;
-import frc.robot.config.Configurable;
-import org.junit.Test;
-
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import frc.robot.config.Config;
+import frc.robot.config.Configurable;
 
 public class ConfigTest {
     /**
@@ -15,17 +16,18 @@ public class ConfigTest {
     @Test
     public void configLoadTest() {
         Config config = new Config();
-        boolean success = config.loadFromFile(getClass().getResource("test_config.txt").getPath());
+        boolean success = config.loadFromFile(getTestFileUrl());
         assertTrue(success);
     }
 
     /**
-     * Tests whether {@link Config} values are parsed and returned as the correct types
+     * Tests whether {@link Config} values are parsed and returned as the correct
+     * types
      */
     @Test
     public void configTypeTest() {
         Config config = new Config();
-        config.loadFromFile(getClass().getResource("test_config.txt").getPath());
+        config.loadFromFile(getTestFileUrl());
         assertEquals("string", config.getSetting("string_value", "wrong string"));
         assertEquals(5.5, config.getSetting("double_value", 1.0), 0.0001);
         assertTrue(config.getSetting("boolean_value", false));
@@ -39,33 +41,35 @@ public class ConfigTest {
     @Test
     public void configCommentTest() {
         Config config = new Config();
-        config.loadFromFile(getClass().getResource("test_config.txt").getPath());
+        config.loadFromFile(getTestFileUrl());
         assertEquals(20.0, config.getSetting("comment_value", 30.0), 0.0001);
     }
 
     /**
-     * Tests whether default values are properly returned for nonexistent {@link Config} values
+     * Tests whether default values are properly returned for nonexistent
+     * {@link Config} values
      */
     @Test
     public void configDefaultValueTest() {
         Config config = new Config();
-        config.loadFromFile(getClass().getResource("test_config.txt").getPath());
+        config.loadFromFile(getTestFileUrl());
         assertEquals("default value", config.getSetting("nonexistent_value", "default value"));
     }
 
     /**
-     * Tests that {@link Configurable Configurables} are called after {@link Config#reloadConfig()}
+     * Tests that {@link Configurable Configurables} are called after
+     * {@link Config#reloadConfig()}
      */
     @Test
     public void configurableTest() {
         Config config = new Config();
-        boolean[] success = {false};
+        boolean[] success = { false };
         Configurable configurable = (inConfig) -> {
             assertEquals(5.5, inConfig.getSetting("double_value", 1.0), 0.0001);
             success[0] = true;
         };
         config.addConfigurable(configurable);
-        config.loadFromFile(getClass().getResource("test_config.txt").getPath());
+        config.loadFromFile(getTestFileUrl());
         assertFalse(success[0]);
         config.reloadConfig();
         assertTrue(success[0]);
@@ -88,5 +92,13 @@ public class ConfigTest {
     public void noConfigTest() {
         Config config = new Config();
         assertFalse(config.loadFromFile("file that i hope never exists"));
+    }
+
+    public String getTestFileUrl() {
+        try {
+            return getClass().getResource("test_config.txt").toURI().getPath();
+        } catch (Exception e) {
+            return "";
+        }
     }
 }

